src/rgw/rgw_http_client.cc

```diff
@@ -226,6 +226,13 @@ static curl_slist *headers_to_slist(param_vec_t& headers)
   return h;
 }
 
+static bool is_upload_request(const char *method)
+{
+  return method == nullptr
```

Comment:
@cbodley should we treat method == nullptr as upload? what does it even mean if it's null?